File: game/IdleMarket/Assets/Scripts/Services/AuthService.cs
using UnityEngine;
using System.Collections;
using System;

//DTOs
public class LoginRequest { public string username; public string password; }
public class LoginResponse { public string token; }

public static class AuthService
{
    public static IEnumerator Login(string username, string password, Action onSuccess, Action<ApiError> onError)
    {
        LoginRequest body = new() { username = username, password = password };

        yield return ApiClient.Post<LoginResponse>("/api/auth/login", body, response =>
        {
            ApiClient.Token = response.token;
            PlayerPrefs.SetString("auth_token", response.token);
            onSuccess?.Invoke();
        }, 
        onError);
    }
}

File: web/src/data/authService.ts
import { clearToken, post, setToken } from "./api";

type LoginResponse = { token: string }

export async function login(username: string, password: string): Promise<void> {
  const data = await post<LoginResponse>("/auth/login", { username, password })

  setToken(data.token)
}

export async function register(username: string, password: string): Promise<void> {
  await post<void>("/auth/register", { username, password })
}

export async function logout(): Promise<void> {
  clearToken()
}